api/domain/eventdata/cluster.proto

```diff
@@ -17,44 +17,47 @@ syntax = "proto3";
 // This file follows google's gRPC naming conventions:
 // https://cloud.google.com/apis/design/naming_convention
 
+import "validate/validate.proto";
+
 option go_package = "github.com/finleap-connect/monoskope/pkg/api/domain/eventdata";
 
 package eventdata;
 
 message ClusterCreated {
   // Display name of the cluster
-  string name = 1;
+  string name = 1 [(validate.rules).string.max_bytes = 150];
   // Unique name of the cluster, to be utilized for generating unique labels
   // and symbols, e.g. with metrics.
-  string label = 2;
+  string label = 2 [(validate.rules).string = {pattern: "^[a-zA-Z_]+$", max_bytes: 60}];
   // Address of the KubeApiServer
-  string api_server_address = 3;
+  string api_server_address = 3 [(validate.rules).string.hostname = true];
   // CA certificate of the K8s cluster
   bytes ca_certificate_bundle = 4;
 }
 
 message ClusterCreatedV2 {
   // Unique name of the cluster, to be utilized for generating unique labels
   // and symbols, e.g. with metrics.
-  string name = 1;
+  string name = 1 [(validate.rules).string = {pattern: "^[a-zA-Z_]+$", max_bytes: 60}];
   // Display name of the cluster
-  string display_name = 2;
+  string display_name = 2 [(validate.rules).string.max_bytes = 150];
   // Address of the KubeApiServer
-  string api_server_address = 3;
+  string api_server_address = 3 [(validate.rules).string.hostname = true];
```

Comment:
Does it allow to validate with schema and port, e.g. "https://k8s-api.lab.example.com:6443"?

Reply:
yap this should not be a problem according to RFC 1034 and if you look [here](https://github.com/envoyproxy/protoc-gen-validate/blob/74a9a1f7b7a5158f65c2dc5c200e1f260619ec5d/validate/validate.h#L112) I see nothing against having a port there. However while checking that I noticed, that hostname doesn't allow for IPs therefore I changed it everywhere to address, which allows for both